digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -0,0 +1,132 @@
+{
+    private static Logger logger = LoggerFactory.getLogger(TdWaitOperatorFactory.class);
+
+    private final TemplateEngine templateEngine;
+
+    @Inject
+    public TdWaitOperatorFactory(TemplateEngine templateEngine, Config systemConfig)
+    {
+        super(systemConfig);
+        this.templateEngine = templateEngine;
+    }
+
+    public String getType()
+    {
+        return "td_wait";
+    }
+
+    @Override
+    public Operator newTaskExecutor(Path workspacePath, TaskRequest request)
+    {
+        return new TdWaitOperator(workspacePath, request);
+    }
+
+    private class TdWaitOperator
+            extends BaseOperator
+    {
+        private TdWaitOperator(Path workspacePath, TaskRequest request)
+        {
+            super(workspacePath, request);
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            Config params = request.getConfig().mergeDefault(
+                    request.getConfig().getNestedOrGetEmpty("td"));
+
+            int pollInterval = getPollInterval(params);
+
+            String query = templateEngine.templateCommand(workspacePath, params, "query", UTF_8);
+
+            int defaultRows = query.trim().isEmpty() ? 0 : 1;
+            int rows = params.get("rows", int.class, defaultRows);
+
+            boolean done = runPollQuery(request, params, query, rows);
+
+            if (done) {
+                return TaskResult.empty(request);
+            }
+            else {
+                throw TaskExecutionException.ofNextPolling(pollInterval, ConfigElement.copyOf(request.getLastStateParams()));
+            }
+        }
+    }
+
+    private boolean runPollQuery(TaskRequest request, Config params, String query, int rows)
+    {
+        String engine = params.get("engine", String.class, "presto");
+        if (!engine.equals("presto") && !engine.equals("hive")) {
+            throw new ConfigException("Unknown 'engine:' option (available options are: hive and presto): " + engine);
+        }
+
+        try (TDOperator op = TDOperator.fromConfig(params)) {
+
+            int priority = params.get("priority", int.class, 0);  // TODO this should accept string (VERY_LOW, LOW, NORMAL, HIGH VERY_HIGH)
+            int jobRetry = params.get("job_retry", int.class, 0);
+
+            TDJobRequest req = new TDJobRequestBuilder()
+                    .setType(engine)
+                    .setDatabase(op.getDatabase())
+                    .setQuery(query)
+                    .setRetryLimit(jobRetry)
+                    .setPriority(priority)
+                    .setScheduledTime(request.getSessionTime().getEpochSecond())
+                    .createTDJobRequest();
+
+            TDJobOperator job = op.submitNewJob(req);
+            logger.info("Started {} job id={}:\n{}", engine, job.getJobId(), query);
+
+            joinJob(job);
+
+            // TODO: wrap query in a query using count and limit instead of storing and downloading the actual full query result?
```

Comment:
I think there are 2 ideas.

(A) td_wait operator always checks that the query returns at least 1 row and the first column of the row is not false or NULL. And users use td_wait as following:
* if you want to check existence of some rows (at least 1 row), use `select true from tbl where [CONDITIONS] limit 1`.
* if you want to check number of rows (at least N rows), use `select count(*) > 100 from tbl where [CONDITIONS]`.

(B) td_wait operator requires `rows: N` option (default: 1). Users will set a query that returns some rows like: `select * where [CONDITIONS]`.
* if N is 1, td_wait operator wraps the query in `select true from (ORIGINAL_QUERY) sub limit 1`
* if N is larger than 1, td_wait operator wraps the query in `select count(*) >= N from (select 1 from (ORIGINAL_QUERY) raw limit N) sub`

(A) is more explicit. (B) is more implicit and has more opportunities to optimize automatically.
I was assuming (A) and this implementation is toward (B).